File: 575-distribute-candies/distribute-candies.cpp
class Solution {
public:
    int distributeCandies(vector<int>& candyType) {
        set<int> newset(candyType.begin(),candyType.end());
        int s = newset.size();
        int n = candyType.size() / 2;

        return min(s,n);
        
    }
};

File: 575-distribute-candies/distribute-candies.py
class Solution(object):
    def distributeCandies(self, candyType):
        """
        :type candyType: List[int]
        :rtype: int
        """
        newset=set(candyType)
        s=len(newset)
        n=len(candyType)//2
        
        return min(s,n)
        